### ISSUE TITLE
docs Scikit-learn examples not working

### ISSUE THREAD

**Issue by chepo92** [kymatio/kymatio#760]
From https://www.kymat.io/codereference.html#module-kymatio.sklearn
I run ` from kymatio.sklearn import Scattering1D, Scattering2D, HarmonicScattering3D`
And then run code example, eg 
```
# Set the parameters of the scattering transform.
J = 3
M, N, O = 32, 32, 32

# Generate a sample signal.
x = np.random.randn(np.prod(M, N, O))

# Define a HarmonicScattering3D object.
S = HarmonicScattering3D(J, (M, N, O))

# Calculate the scattering transform.
Sx = S.scattering(x)

# Equivalently, use the alias.
Sx = S.predict(x)
```
none of the examples work, errors throwed: 
3D:
```
TypeError                                 Traceback (most recent call last)

<ipython-input-65-7d3625924daf> in <module>()
      4 
      5 # Generate a sample signal.
----> 6 x = np.random.randn(np.prod(M, N, O))
      7 
      8 # Define a HarmonicScattering3D object.

<__array_function__ internals> in prod(*args, **kwargs)

1 frames

/usr/local/lib/python3.7/dist-packages/numpy/core/fromnumeric.py in _wrapreduction(obj, ufunc, method, axis, dtype, out, **kwargs)
     85                 return reduction(axis=axis, out=out, **passkwargs)
     86 
---> 87     return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
     88 
     89 

TypeError: Cannot interpret '32' as a data type
```
1D:
```
TypeError                                 Traceback (most recent call last)

<ipython-input-66-79e7ca25bbde> in <module>()
     14 
     15 # Equivalently, use the alias.
---> 16 Sx = S.predict(x)

/usr/local/lib/python3.7/dist-packages/kymatio/frontend/sklearn_frontend.py in predict(self, x)
      9     def predict(self, x):
     10         n_samples = x.shape[0]
---> 11         x = x.reshape((-1,) + self.shape)
     12         Sx = self.scattering(x)
     13         Sx = Sx.reshape(n_samples, -1)

TypeError: can only concatenate tuple (not "int") to tuple

```

2D: 

```
AxisError                                 Traceback (most recent call last)

<ipython-input-64-d0dc2efe80b0> in <module>()
      4 
      5 # Generate a sample signal.
----> 6 x = np.random.randn(np.prod(M, N))
      7 
      8 # Define a Scattering2D object.

<__array_function__ internals> in prod(*args, **kwargs)

1 frames

/usr/local/lib/python3.7/dist-packages/numpy/core/fromnumeric.py in _wrapreduction(obj, ufunc, method, axis, dtype, out, **kwargs)
     85                 return reduction(axis=axis, out=out, **passkwargs)
     86 
---> 87     return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
     88 
     89 

AxisError: axis 32 is out of bounds for array of dimension 0
```

System: 
```
Linux-5.4.104+-x86_64-with-Ubuntu-18.04-bionic
Python 3.7.11 (default, Jul  3 2021, 18:01:19) 
[GCC 7.5.0]
NumPy 1.19.5
SciPy 1.4.1
PyTorch 1.9.0+cu102
Kymatio 0.2.1
```

**Comment by MuawizChaudhary** [comment]
The example is very wrong. Will submit a PR in a bit. Thanks @chepo92

**Comment by janden** [comment]
Was solved by #823.
